### PR TITLE
Add support for Ubuntu 15.04 in docker-multinode cluster setup

### LINKED ISSUE #14429: docker-multinode does not work on Ubuntu 15.04
The problem is that docker is managed by systemd and so the configuration files which are written in master.sh and worker.sh are ignored.

I have a change locally that fixes this, but need to polish it.

@resouer @dalanlan @brendandburns

### PR DESCRIPTION
**Pull request by fgrzadkowski**
Fixes #14429 #8798

@brendandburns @dalanlan @resouer

### CODE REVIEW

File: docs/getting-started-guides/docker-multinode/master.sh [hunk 1]
```diff
@@ -105,28 +104,37 @@ start_k8s(){
...
-        fedora|centos|amzn)
-            DOCKER_CONF="/etc/sysconfig/docker"
+        fedora*|centos*|amzn*)
+            echo "DOCKER_OPTS=\"\$DOCKER_OPTS --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}\"" | sudo tee -a /etc/sysconfig/docker
+        ;;
+        ubuntu15.04)
+            install -Dv /dev/null /etc/systemd/system/docker.service.d/bridge.conf
+            cat <<EOF > /etc/systemd/system/docker.service.d/bridge.conf
+[Service]
+ExecStart=
```
Comment (mikedanese) [inline docs/getting-started-guides/docker-multinode/master.sh]: is this empty ExecStart required?

File: docs/getting-started-guides/docker-multinode/master.sh [hunk 2]
```diff
@@ -59,7 +59,7 @@ detect_lsb() {
     esac
 
     if command_exists lsb_release; then
-        lsb_dist="$(lsb_release -si)"
+        lsb_dist="$(lsb_release -sir | sed ':a;N;s/\n//g')"
```
Comment (mikedanese) [inline docs/getting-started-guides/docker-multinode/master.sh]: why is the `:a` label required?

File: docs/getting-started-guides/docker-multinode/master.sh [hunk 3]
```diff
@@ -75,6 +75,7 @@ detect_lsb() {
     fi
 
     lsb_dist="$(echo ${lsb_dist} | tr '[:upper:]' '[:lower:]')"
+    echo "Detected $lsb_dist"
```
Comment (mikedanese) [inline docs/getting-started-guides/docker-multinode/master.sh]: At this point lsb_dist could still be `""`. Is that an issue?

Comment (dalanlan) [inline docs/getting-started-guides/docker-multinode/master.sh]: If that's the case, it definitely is. 
Yet currently we only support limited platforms (debian, ubuntu, centos, amzn). If this works for them all, i believe it's fine for now. (should be well doc though)
The original idea of this lsb_release detection cames from docker installation guide, FYI.

File: docs/getting-started-guides/docker-multinode/master.sh (hunk 1 above)
Comment (fgrzadkowski) [inline docs/getting-started-guides/docker-multinode/master.sh]: Yes. https://wiki.archlinux.org/index.php/Systemd#Drop-in_snippets

File: docs/getting-started-guides/docker-multinode/master.sh (hunk 2 above)
Comment (fgrzadkowski) [inline docs/getting-started-guides/docker-multinode/master.sh]: Please see http://stackoverflow.com/a/1252191 for explanation

**Comment by mikedanese** [comment]
LGTM

File: docs/getting-started-guides/docker-multinode/master.sh (hunk 3 above)
Comment (mikedanese) [inline docs/getting-started-guides/docker-multinode/master.sh]: should be fixed but the bug was not introduced in this PR so we can fix in a followup

Comment (dalanlan) [inline docs/getting-started-guides/docker-multinode/master.sh]: /cc @resouer who's the original author

**Comment by a-robinson** [comment]
Needs a rebase before it can merged, unfortunately

File: docs/getting-started-guides/docker-multinode/master.sh (hunk 3 above)
Comment (resouer) [inline docs/getting-started-guides/docker-multinode/master.sh]: I did ... Would u mind to send a PR to fix this in other place :) @dalanlan

Comment (dalanlan) [inline docs/getting-started-guides/docker-multinode/master.sh]: Sure

File: docs/getting-started-guides/docker-multinode/master.sh [hunk 4]
```diff
@@ -105,28 +104,37 @@ start_k8s(){
...
     ifconfig docker0 down
 
     case "$lsb_dist" in
-        fedora|centos|amzn)
+        fedora*|centos*)
             yum install bridge-utils && brctl delbr docker0 && systemctl restart docker
         ;;
-        ubuntu|debian|linuxmint)
+        ubuntu*|debian*|linuxmint*)
             apt-get install bridge-utils && brctl delbr docker0 && service docker restart
```
Comment (resouer) [inline docs/getting-started-guides/docker-multinode/master.sh]: Is ubuntu15.04 still use `service docker restart`?

**Comment by piosz** [comment]
@fgrzadkowski could you please rebase?

**Comment by wojtek-t** [comment]
@fgrzadkowski  - please rebase

**Comment by wojtek-t** [comment]
@fgrzadkowski - please rebase

**Comment by erictune** [comment]
@fgrzadkowski - please rebase

**Comment by fgrzadkowski** [comment]
I'm approaching this problem from a different angle now, so I'm gonna close this PR and reopen if needed.

**Pull request closed without merging**